ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx

```diff
@@ -51,6 +52,14 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
         onBlur={handleQueryBlur}
         margin="dense"
       />
+      <TextField
+        fullWidth
+        label="Series Name Format"
+        value={format ?? ''}
+        onChange={handleFormatChange}
+        onBlur={handleFormatBlur}
+        margin="dense"
+      />
```

Comment:
I think I remember in the designs that Mohit called the field "Legend" and then there was an info tooltip to explain more about the formatting syntax

Reply:
We could always add the info tooltip in a follow-up PR

Reply:
Ok, looking at the screenshot ... part of me also wants to move this input to the section that's titled "Legend"

Reply:
We discussed this more offline but to summarize, the screenshot in the description only has one query but when there are multiple, each query needs to specify how series names should be formatted (some good discussion on #600 about this). There is potential that further customization is possible from the chart-level legend in the future though! Also, I'm following up about what should go in the TextField label / info tooltip